trace-dispatcher/doc/trace-dispatcher.md

```diff
@@ -519,30 +570,74 @@ The self-documentation features of `trace-dispatcher` are provided by a combinat
 
-*Because it is not enforced by the type system, it is very important that each trace provides a complete list of `DocMsg` entries for all message contructors, as these prototypes are also used for configuration*.
+TODO: Add consistency check between configuration and documentation.
 
-If a documentation entry is missing, the configuration system will not work as expected.
-In a future release we want to enhance the library, so that the completeness will be checked by the Haskell type checker.
 
 ```haskell
 newtype Documented a = Documented {undoc :: [DocMsg a]}
 
 data DocMsg a = DocMsg {
-    dmPrototype :: a
+    dmNamespace :: Namespace
   , dmMetricsMD :: [(Text, Text)]
   , dmMarkdown  :: Text
 }
 ```
 
+## Metrics
+
+Metrics are provided by normal trace messages, which implement the `asMetrics` function
+of the `LogFormatting` typeclass. For this reason all the configuration mechanisms for
+filtering and routing can be used with metrics. `ekgTracer`is used as the metrics backend.
+It forwards the metrics to cardano-tracer for further processing.  
+
+## DataPoints
+
+DataPoints gives the ability for processes other then cardano-node to query the provided
+runtime state of a node. DataPoints are equal to metrics, in that they are not written in textual
+form to a log, but in contrast to metrics they have an ADT structure, so they can trace
+any structured information. As a result, they give the ability for external processes
+other then cardano-node to query the provided runtime state of a node (for example,
+node's basic information).
+
+DataPoints are implemented as special tracers, which packs the objects into DataPoint
+constructors and require a ToJSON instance for that objects. The set of DataPoints
+provided by the node is structured using the same namespace as metrics and log messages.
+But otherwise DataPoints work independent of tracing, but are written in a local store,
+so the latest value of a particular DataPoint can be queried on demand.
+
+Also, [there is a document](https://github.com/input-output-hk/cardano-node/wiki/cardano-node-and-DataPoints:-demo)
+describing how to accept DataPoints from an external process.
+
+[`demo-acceptor`](https://github.com/input-output-hk/cardano-node/blob/master/cardano-tracer/demo/acceptor.hs)
+application allows to ask for particular DataPoint by its name and display its value.
+
+```haskell
+-- A simple dataPointTracer which supports building a namespace
+-- `trDataPoint`: shall be the datapoint backend
+-- `namesFor`: shall be a function to produce a namespace for every
+-- datapoint constructor
+mkDataPointTracer :: forall dp. ToJSON dp
+  => Trace IO DataPoint
+  -> (dp -> [Text])
+  -> IO (Trace IO dp)
+mkDataPointTracer trDataPoint namesFor = do
+    let tr = NT.contramap DataPoint trDataPoint
+    pure $ withNamesAppended namesFor tr```
```

Comment:
Stray triple back ticks at end of above line?

Reply:
Fixed